### PR TITLE
export jsf-2.2 impl classes as third-party

### LINKED ISSUE #8521: Re-introduce change reverted from 8041
Re-introduce change reverted from https://github.com/OpenLiberty/open-liberty/pull/8041

### PR DESCRIPTION
**Pull request by wtlucy**
This PR un-reverts #7768, and additionally limits the scope of the org.apache.myfaces packages that are made available to third-party API classloaders.

fixes #8521 and #7767

### CODE REVIEW

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.2_fat

File: dev/com.ibm.websphere.appserver.jsfApiStub-2.2/.project [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
Comment (pnicolucci) [inline dev/com.ibm.websphere.appserver.jsfApiStub-2.2/.project]: Are these gradle entries required?

Comment (wtlucy) [reply dev/com.ibm.websphere.appserver.jsfApiStub-2.2/.project]: @jhanders34 didn't call this out when he first fixed the eclipse projects settings for this project (pre-revert) in #8343, so I doubt it's a problem.  Do you have an eclipse environment you can check this out in?

Comment (wtlucy) [reply dev/com.ibm.websphere.appserver.jsfApiStub-2.2/.project]: I actually can't find any other projects using that `org.eclipse.buildship.core.gradleprojectnature` - I'll just delete it.

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.2_fat

**Comment by wtlucy** [comment]
A failure in com.ibm.ws.security.openidconnect.client:test killed the last personal build, so I've just pushed a rebase on integration.

#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.2_fat

**Comment by wtlucy** [comment]
Previous build failed for no obvious reason, cool.

#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.2_fat

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.2_fat

**Pull request merged**